api/envoy/config/core/v3/health_check.proto

```diff
@@ -354,4 +354,15 @@ message HealthCheck {
   // the cluster's :ref:`transport socket <envoy_api_field_config.cluster.v3.Cluster.transport_socket>`
   // will be used for health check socket configuration.
   google.protobuf.Struct transport_socket_match_criteria = 23;
+
+  // The "no traffic healthy interval" is a special health check interval that is used when a cluster has
+  // never had traffic routed to it but is a healthy cluster.
```

Comment:
From reading the code I can tell that it's not about the health of the cluster (which isn't a formalized concept in Envoy to begin with), but rather with the health of the individual host. It also is actually checking whether active health checking has failed or not, which is just one source of health information (EDS is another one).

How about `... is used for hosts that are currently failing active health checking (including new hosts) when the cluster has received no traffic.`?